R/library.R

```diff
@@ -191,11 +154,102 @@ current_version.locked_package <- function(package) {
   current_version(package$name)
 }
 
-description_file_for <- function(package_name) {
-  dcf_file <- file.path(libPath(), package_name, "DESCRIPTION")
+description_file_for <- function(package_name, libP) {
+  dcf_file <- file.path(libP, package_name, "DESCRIPTION")
   if (file.exists(dcf_file)) {
     read.dcf(dcf_file)
   } else {
     NULL
   }
 }
+
+download_package <- function(package) {
+  UseMethod("download_package")
+}
+
+#' Download CRAN package, either current or older version
+download_package.CRAN <- function(package) {
+  remote_version <- get_available_cran_version(package)
+  name <- package$name
+  version <- package$version
+  repo <- "http://cran.r-project.org"
+
+  ## Dependency package means that we grab the latest.  download.packages
+  ## does a better job grabbing the latest than we do.  Edge-cases where certain
+  ## things have been updated on the information page for a package but others
+  ## have not.
+  if (package$is_dependency_package) {
+    pkg_tarball <- tempfile(fileext = ".tar.gz")
```

Comment:
I was doing something similar at one point, then decided to make use of temp :(
Will refactor